AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -224,9 +224,13 @@ - (void)_layoutNodes:(NSArray *)nodes ofKind:(NSString *)kind atIndexPaths:(NSAr
 
   // Block the _editingTransactionQueue from executing a new edit transaction until layout is done & _editingNodes array is updated.
   dispatch_group_wait(layoutGroup, DISPATCH_TIME_FOREVER);
-  free(nodeBoundSizes);
-
+  
   if (completionBlock) {
+    NSMutableArray *indexPaths = [NSMutableArray arrayWithCapacity:nodeCount];
+    [contexts enumerateObjectsUsingBlock:^(ASIndexedNodeContext * _Nonnull context, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
It would be a little faster here to use NSFastEnumeration with a for-loop. That way we avoid the dynamic method call and allocating/invoking the dispatch block.